color_picker/src/main/ets/components/MainPage/colorPicker.ets

```diff
@@ -0,0 +1,211 @@
+    cancel: this.existApp,
+    autoCancel: true,
+    alignment: DialogAlignment.Bottom,
+  });
+
+  onAccept(value: number) {
+    this.OpacityOne = 1
+    this.hue = value;
+    this.SliderSaturation = 100
+    this.SliderLightness = 50
+    var RGBPixels = this.HSLToRGB(this.hue, this.SliderSaturation, this.SliderLightness)
+    var HEXCode = this.rgbToHex(RGBPixels[0], RGBPixels[1], RGBPixels[2])
+    this.ColorFromS = HEXCode
+    this.ColorFromGradient = HEXCode
+    this.FinalColor = HEXCode
+  }
+
+  existApp() {
+    this.OpacityOne = 1
+    console.log("Cancel dialog!");
+  }
+
+  rgbToHex(r, g, b) {
+    return "#" + ((1 << 24) + (r << 16) + (g << 8) + b).toString(16).slice(1);
+  }
+
+  HSLToRGB(h, s, l) {
+    s /= 100;
+    l /= 100;
+    const k = n => (n + h / 30) % 12;
+    const a = s * Math.min(l, 1 - l);
+    const f = n =>
+    l - a * Math.max(-1, Math.min(k(n) - 3, Math.min(9 - k(n), 1)));
+    return [Math.round(255 * f(0)), Math.round(255 * f(8)), Math.round(255 * f(4))];
+  }
+
+  build() {
+    Column() {
+      Column() {
+        Row() {
+          Flex({ justifyContent: FlexAlign.SpaceEvenly }) {
+            Button('Selection of Color')
+              .width('250vp')
+              .height('50vp')
+              .fontSize('22fp')
+              .backgroundColor(Color.White)
+              .fontColor(this.FinalColor)
+              .fontWeight(FontWeight.Bolder)
+              .onClick(() => {
+                this.OpacityOne = 0.2
+                this.dialogController.open()
+              })
+
+            Button('OK')
+              .fontColor(Color.Black)
+              .fontWeight(FontWeight.Bolder)
+              .backgroundColor(0XE0E1E4)
+              .onClick(() => {
+                console.log(this.FinalColor)
+                router.back()
+              })
+          }
+        }
+        .backgroundColor('#c2b7bf')
+        .height('70vp')
+        .borderRadius('20vp')
+
+        Row()
+          .height('20vp')
+        Row()
+          .height('40%')
+          .width('340vp')
+          .linearGradient({
+            angle: 90,
+            direction: GradientDirection.Left,
+            colors: [[Color.Red, 0], [Color.Yellow, 1 / 6], [0x00FF00, 2 / 6], [0x00FFFF, 3 / 6], [Color.Blue, 4 / 6], [0xFF00FF, 5 / 6], [Color.Red, 1]]
+          })
+          .onClick((event: ClickEvent) => {
+            this.hue = (event.x / 340) * 360;
+            this.SliderSaturation = 100
+            this.SliderLightness = 50
+            var RGBPixels = this.HSLToRGB(this.hue, this.SliderSaturation, this.SliderLightness)
+            var HEXCode = this.rgbToHex(RGBPixels[0], RGBPixels[1], RGBPixels[2])
+            this.FinalColor = HEXCode
+            this.ColorFromGradient = HEXCode
+            this.ColorFromS = HEXCode
+          })
+        Row()
+          .height('40vp')
+        Flex({ justifyContent: FlexAlign.Start }) {
+          Text("Saturation :").fontWeight(FontWeight.Bolder).fontSize('15fp')
```

Comment:
Saturation and lightness are part of base library? If not either remove or make then configurable that provide an option to user to decide if these need to be displayed or not